### PR TITLE
Automatically attach packaged deployment YAMLs to GitHub Release

### LINKED ISSUE #764: Automatically attach packaged deployment YAMLs to GitHub Release
As per #739 we are generating packaged version of our deployment YAMLs but they are not automatically linked to release and documented in our release process.

However, this is error prone and was forgotten for v1.4 (see #760) so we should add a step to our release flow to automatically do this.

### PR DESCRIPTION
**Pull request by tomkerkhove**
Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

Automatically attach packaged deployment YAMLs to GitHub Release as part of our release workflow.

### Checklist

- [x] ~A PR is opened to update the documentation on https://github.com/kedacore/keda-docs~
- [X] Commits are signed with Developer Certificate of Origin (DCO)
- [x] ~Tests have been added~
- [X] Manual test was done in separate repo (see [release](https://github.com/tomkerkhove/keda-sandbox/releases/tag/v1.2.1) and [workflow](https://github.com/tomkerkhove/keda-sandbox/runs/684800356?check_suite_focus=true))
- [X] Update release process documentation

Fixes #764

### CODE REVIEW

**Comment by tomkerkhove** [comment]
Currently, we advise triggering the process by creating a GitHub release (which creates a Git tag).

However, I'm thinking that we should maybe change this so that we just create a tag and we manage the GitHub release in the workflow as well so that we have consistent structure in our releases. Thoughts?

File: .github/workflows/release-build.yml [hunk 1]
```diff
@@ -31,3 +31,34 @@ jobs:
...
+
+      # Upload deployment YAML files as TAR to GitHub release
+      - name: Upload Deployment YAML (zip)
+        id: upload-deployment-yaml-tar
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: https://uploads.github.com/repos/tomkerkhove/keda-sandbox/releases/${{ steps.get-release-info.outputs.id }}/assets?name=keda-1.4.1.tar.gz
+          asset_path: keda-1.4.1.tar.gz
```
Comment (tomkerkhove) [inline .github/workflows/release-build.yml:62]: @zroubalik I'll need help with this, where does MAKE output these?

File: .github/workflows/release-build.yml [hunk 2]
```diff
@@ -31,3 +31,34 @@ jobs:
...
+
+      # Upload deployment YAML files as ZIP to GitHub release
+      - name: Upload Deployment YAML (zip)
+        id: upload-deployment-yaml-zip
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: https://uploads.github.com/repos/tomkerkhove/keda-sandbox/releases/${{ steps.get-release-info.outputs.id }}/assets?name=keda-1.4.1.zip
+          asset_path: keda-1.4.1.zip
```
Comment (tomkerkhove) [inline .github/workflows/release-build.yml:50]: @zroubalik I'll need help with this, where does MAKE output these?

File: .github/workflows/release-build.yml (hunk 1 above)
Comment (zroubalik) [reply .github/workflows/release-build.yml:62]: to the root of the project eg `./keda-1.4.1.tar.gz`

File: .github/workflows/release-build.yml (hunk 2 above)
Comment (zroubalik) [reply .github/workflows/release-build.yml:50]: to the root of the project

File: .github/workflows/release-build.yml (hunk 1 above)
Comment (tomkerkhove) [reply .github/workflows/release-build.yml:62]: Then we should be good to go, was same on https://github.com/tomkerkhove/keda-sandbox

File: .github/workflows/release-build.yml (hunk 2 above)
Comment (tomkerkhove) [reply .github/workflows/release-build.yml:50]: Then we should be good to go, was same on https://github.com/tomkerkhove/keda-sandbox

Comment (zroubalik) [inline .github/workflows/release-build.yml:50]: Is it intentional that the version (1.4.1) is hardcoded in here?

Comment (zroubalik) [reply .github/workflows/release-build.yml:50]: as well as the uploads ulr contains `tomkerhove` repo :)

Comment (tomkerkhove) [reply .github/workflows/release-build.yml:50]: Ha, nope 🙈

Comment (tomkerkhove) [reply .github/workflows/release-build.yml:50]: I'll fix this asap but don't have any more time today/this week.

Seems like getting the version out of the tag and setting it as a variable is not a trivial thing 🙄

Comment (ahmelsayed) [reply .github/workflows/release-build.yml:50]: I think https://github.com/kedacore/keda/pull/832 should simplify this for you. It's based on this [approach](https://github.community/t5/GitHub-Actions/How-to-get-just-the-tag-name/m-p/32167/highlight/true#M1027). I tried to use it before, but couldn't get it to work, though later realized that I was missing the env section on the tasks that need it set.

Comment (tomkerkhove) [reply .github/workflows/release-build.yml]: That did the trick, thanks @ahmelsayed!

**Comment by tomkerkhove** [comment]
New example: https://github.com/tomkerkhove/keda-sandbox/releases/tag/v2.0.0

**Pull request merged**